src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCache.cs

```diff
@@ -67,27 +79,26 @@ public CompilerCache(IFileProvider fileProvider)
         /// <inheritdoc />
         public CompilerCacheResult GetOrAdd(
```

Comment:
Would be better to return `Task<>` here and let the caller decide if they should block or not?